### PR TITLE
Add `BinaryData.isReplayable`

### PR DESCRIPTION
**Pull request by kasobol-msft**
This PR;

- Adds `BinaryData.isReplayable` that returns `true` for contents that are known to be always replayable today.

Out of scope.
- The `BinaryData.toReplayableData`/ `BinaryData.toReplayableDataAsync`  (or so) will be separate work. The `BinaryData.isReplayable` is MVP to get file transfers optimized. (see https://github.com/Azure/azure-sdk-for-java/issues/28799 )

### CODE REVIEW

File: sdk/core/azure-core/src/main/java/com/azure/core/implementation/util/InputStreamContent.java [hunk 1]
```diff
@@ -80,6 +80,11 @@ public Flux<ByteBuffer> toFluxByteBuffer() {
         return FluxUtil.toFluxByteBuffer(this.content, STREAM_READ_SIZE);
     }
 
+    @Override
+    public boolean isReplayable() {
+        return false;
```
Comment (srnagar) [inline sdk/core/azure-core/src/main/java/com/azure/core/implementation/util/InputStreamContent.java:85]: Should this be based on the type of input stream? If the source is ByteArrayInputStream, this should return true, for e.g.

File: sdk/core/azure-core/src/main/java/com/azure/core/util/BinaryData.java [hunk 2]
```diff
@@ -1459,4 +1459,13 @@ public Flux<ByteBuffer> toFluxByteBuffer() {
     public Long getLength() {
         return content.getLength();
     }
+
+    /**
+     * Returns a flag indicating whether the content can always be repeatedly consumed using all accessors including
+     * {@link #toStream()} and {@link #toFluxByteBuffer()}
+     * @return a flag indicating whether the content can always be repeatedly consumed using all accessors.
```
Comment (srnagar) [inline sdk/core/azure-core/src/main/java/com/azure/core/util/BinaryData.java:1466]: nit: We should probably not claim that it'll "always" be available. For e.g. for FileContent, we don't read the file contents into memory eagerly and rely on the file to not be deleted to be replayable.

**Review by srnagar — APPROVED** [review]

File: sdk/core/azure-core/src/main/java/com/azure/core/implementation/util/InputStreamContent.java (hunk 1 above)
Comment (kasobol-msft) [reply sdk/core/azure-core/src/main/java/com/azure/core/implementation/util/InputStreamContent.java:85]: That's going to be part of https://github.com/Azure/azure-sdk-for-java/issues/28799

**Review by alzimmermsft — APPROVED** [review]

**Review by JonathanGiles — COMMENTED** [review]
I got distracted midway through this review, but I wanted to record the one comment I had made.

**Pull request merged**